### ISSUE TITLE
Best way to use this library

### ISSUE THREAD

**Issue by m8schmit** [mrbungle64/ecovacs-deebot.js#97]
Hi, I don't have a lot of IoT knowledge so I was wondering what is the best way to use this library? 

I first did a test with an app.js file launched directly through Node from my terminal. It works very well! 
Then I tried to install it an Angular webapp and it was immediately more complicated; I had to fix several things to make it work:
- add *.d.ts files
- remove the `fs`library
- change the `url` library

... And I learned I can't have `machine-id` on Angular...

I did another test with React Native, fix the same problem, get a machine-id with `react-native-device-info`,

... And I had CORS problems when I called the Ecovacs API.

So like I said, what is the best way to use this library? Create a second MQTT server between this library and my app maybe?

Thanks in advance for your help

**Comment by mrbungle64** [comment]
@m8schmit

It seems that you want to create a client-side application. Typically, this library is used for server-side applications.

Can you please describe your use case?

There's one client-side application that is using this library:
https://github.com/Dracks/ecovacs-home-desktop-clone
I haven't tried it myself yet, but have you checked it out?

**Comment by m8schmit** [comment]
Thank you for your answer!
I'll take a look at https://github.com/Dracks/ecovacs-home-desktop-clone.

Long story short, I bought a T9 Aivi from China, it works great but makes a little too many requests to aliyuncs/ecovacs/ecouser for my taste ...
I looked at Bumper for the server, but it requires you to root your phone to continue using the official app. As I wasn't too keen on that, I started looking at where to find an alternative app. I tested several SmartHome applications but they are too complete. So I am planning to recreate an application similar to Ecovacs Home and I was thinking of starting from this library to save time.
